src/main/scala/com/normation/plugins/datasources/Repository.scala

```diff
@@ -161,7 +161,31 @@ class DataSourceRepoImpl(
   , uuidGen: StringUuidGenerator
 ) extends DataSourceRepository with DataSourceUpdateCallbacks {
 
-  private[this] var datasources = Map[DataSourceId, DataSourceScheduler]()
+  /*
+   * Be careful, ALL modification to datasource must be synchronized
+   */
+  private[this] object datasources extends AnyRef {
+    private[this] var internal = Map[DataSourceId, DataSourceScheduler]()
+    // utility methods on datasources
+    // stop a datasource - must be called when the datasource still in "datasources"
+    private[this] def stop(id: DataSourceId) = {
+      DataSourceLogger.debug(s"Stopping data source with id '${id.value}'")
+      internal.get(id) match {
+        case None      => s"Data source with id ${id.value} was not found running"
```

Comment:
this text doesn't go anywhere, unless i'm mistaken. Shouldn't it go to a logger ?